### PR TITLE
Implement dissocPath

### PR DESCRIPTION
**Pull request by Hilzu**
Implement `dissocPath` function. I used the implementation of `assocPath` as a starting point and this is quite similar to it. `dissocPath` was mentioned in https://github.com/ramda/ramda/pull/781 and I decided to implement it because I needed it for my own project.

### CODE REVIEW

**Comment by CrossEye** [comment]
LGTM

:herb:

File: src/dissocPath.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * @category Object
+ * @sig String -> {k: v} -> {k: v}
+ * @param {String} path the dot-delimited path to set
+ * @param {Object} obj the object to clone
+ * @return {Object} a new object without the property at path
+ * @example
+ *
+ *      var obj1 = {a: {b: 1, c: 2, d: {e: 3}}, f: {g: {h: 4, i: 5, j: {k: 6, l: 7}}}, m: 8};
+ *      var obj2 = R.dissocPath('f.g.i', obj1);
+ *      //=> {a: {b: 1, c: 2, d: {e: 3}}, f: {g: {h: 4, j: {k: 6, l: 7}}}, m: 8}
```
Comment (davidchambers) [inline src/dissocPath.js]: Could you simplify this example? It's a bit hard to see how the input and output differ.

File: src/dissocPath.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ */
+module.exports = (function() {
+    var setParts = function(parts, obj) {
+        if (parts.length === 1) {return dissoc(parts[0], obj);}
+        var current = obj[parts[0]];
+        return is(Object, current) ? assoc(parts[0], setParts(_slice(parts, 1), current), obj) : obj;
+    };
+    return _curry2(function(path, obj) {
```
Comment (davidchambers) [inline src/dissocPath.js]: You could name this function `dissocPath`.

File: test/dissocPath.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        F.prototype.a = 1;
+        var obj1 = new F();
+        obj1.b = {c: 2, d: 3};
+        var obj2 = R.dissocPath('b.c', obj1);
+        assert.deepEqual(obj2,
+            {a: 1, b: {d: 3}}
+        );
+    });
+
+    it('is properly curried', function() {
```
Comment (davidchambers) [inline test/dissocPath.js]: The word "properly" isn't necessary.

**Comment by buzzdecafe** [comment]
LGTM as well, subject to the following caveats:

1. please squash into one commit; and
2. i still think we should change dotted string paths for `[String]`, but that is beyond the scope of this PR

File: test/dissocPath.js (hunk 3 above)
Comment (CrossEye) [inline test/dissocPath.js]: Is that an improper "properly"?

**Comment by Hilzu** [comment]
All requests have been done and this PR should be ready for merge.

**Comment by buzzdecafe** [comment]
thank you!

**Pull request merged**